osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -113,10 +112,7 @@ public void ApplyDefaults(ControlPointInfo controlPointInfo, BeatmapDifficulty d
             nestedHitObjects.Sort((h1, h2) => h1.StartTime.CompareTo(h2.StartTime));
 
             foreach (var h in nestedHitObjects)
-            {
-                h.HitWindows = HitWindows;
```

Comment:
Just out of curiosity, why'd the propagation of hit windows to the lower nested objects go away?

Reply:
ah, i forgot to mention that. the behaviour wasn’t well tested or documented, and would result in nested hitobjects that create their own hitwindows having them silently overwritten.

smoogi mentioned this was a memory optimisation for difficulty calculation. we can revisit it if it’s an issue later.